pkg/version/release.go

```diff
@@ -15,7 +16,8 @@ type ReleaseVersion struct {
 // - v1.1.0-73-g7e6d4f69
 // - v1.1.0
 func ParseGitVersion(gitVersion string) (*ReleaseVersion, error) {
-	v, err := utilversion.ParseGeneric(gitVersion)
+	filteredVersion := removeGitVersionCommits(gitVersion)
```

Comment:
The `ParseGitVersion` is responsible for parsing a version with `string format` to a structured version type, it doesn't know how this version will be consumed. 

I think the `remove git version` should be delayed to `ReleaseVersion()`.